src/azure-cli-core/azure/cli/core/__init__.py

```diff
@@ -201,24 +206,29 @@ def load_command_table(self, args):
             _load_module_command_loader, _load_extension_command_loader, BLACKLISTED_MODS, ExtensionCommandSource)
         from azure.cli.core.extension import (
             get_extensions, get_extension_path, get_extension_modname)
+        from azure.cli.core._session import INDEX
 
-        def _update_command_table_from_modules(args):
+        def _update_command_table_from_modules(args, command_modules=None):
             '''Loads command table(s)
             When `module_name` is specified, only commands from that module will be loaded.
             If the module is not found, all commands are loaded.
             '''
-            installed_command_modules = []
-            try:
-                mods_ns_pkg = import_module('azure.cli.command_modules')
-                installed_command_modules = [modname for _, modname, _ in
-                                             pkgutil.iter_modules(mods_ns_pkg.__path__)
-                                             if modname not in BLACKLISTED_MODS]
-            except ImportError as e:
-                logger.warning(e)
-
-            logger.debug('Installed command modules %s', installed_command_modules)
+
+            if not command_modules:
+                # Perform module discovery
+                command_modules = []
+                try:
+                    mods_ns_pkg = import_module('azure.cli.command_modules')
+                    command_modules = [modname for _, modname, _ in
+                                       pkgutil.iter_modules(mods_ns_pkg.__path__)
+                                       if modname not in BLACKLISTED_MODS]
+                    logger.debug('Discovered command modules: %s', command_modules)
+                except ImportError as e:
+                    logger.warning(e)
+
+            logger.debug("Loading command modules: %s", command_modules)
             cumulative_elapsed_time = 0
-            for mod in [m for m in installed_command_modules if m not in BLACKLISTED_MODS]:
+            for mod in [m for m in command_modules if m not in BLACKLISTED_MODS]:
```

Comment:
We can remove `if m not in BLACKLISTED_MODS` as it is already done in line 224?

Reply:
Nice catch, but the index may be out-dated and contains a blocked mod. I will remove it from L224 instead.

Reply:
Also, I have changed the name `BLACKLISTED_MODS` to the neutral `BLOCKED_MODS`, as per https://bugs.chromium.org/p/chromium/issues/detail?id=981129